### PR TITLE
--color=debug: combine segments with same labels

### PR DESCRIPTION
**Pull request by jonathantanmy**
I noticed this when doing the diff whitespace work.

(notice the repeated "diff removed line_number" labels) From
```
[38;5;1m<<diff removed line_number:: >><<diff removed line_number:: >><<diff removed line_number:: >><<diff removed line_number::1>>[39m<<diff:: >><<diff::    : >>[4m[38;5;1m<<diff removed token::foo>>[24m[39m
```
to
```
[38;5;1m<<diff removed line_number::   1>>[39m<<diff::     : >>[4m[38;5;1m<<diff removed token::foo>>[24m[39m
```

# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

**Comment by ilyagr** [comment]
We discussed this idea with @yuja previously, and he was leaning against it.

However, I still like the idea, since it's so nice to have readable tests. Yuya, if you still would like to have the debug mode to report precisely what happened, would you perhaps be OK with having a separate debug mode for more readable output?

File: cli/src/formatter.rs [hunk 1]
```diff
@@ -272,6 +272,8 @@ pub struct ColorFormatter<W: Write> {
     cached_styles: HashMap<Vec<String>, Style>,
     /// The style we last wrote to the output.
     current_style: Style,
+    /// The debug string (space-separated labels) we last wrote to the output.
+    current_debug: String,
```
Comment (yuja) [inline cli/src/formatter.rs:276]: nit: for extra compile-time guarantee, maybe we can combine `debug: bool` and `current_debug: String` as `Option<String>` or `current_debug_labels: Option<Vec<String>>`?

File: cli/src/formatter.rs [hunk 2]
```diff
@@ -492,36 +514,19 @@ impl<W: Write> Write for ColorFormatter<W> {
...
-            if debug {
-                write!(output, "<<{}::", labels.join(" "))?;
-                write_sanitized(output, buf)?;
-                write!(output, ">>")?;
-            } else {
-                write_sanitized(output, buf)?;
-            }
-
+        fn write_line_exclusive<W: Write>(output: &mut W, buf: &[u8]) -> io::Result<()> {
+            write_sanitized(output, buf)?;
```
Comment (yuja) [inline cli/src/formatter.rs:529]: nit: inline this

**Review by yuja — APPROVED** [review]
I don't wanna introduce complexity in debug code, which was the main motivation to not deduplicate label outputs. I think this implementation is simple enough to get in. Thanks.

File: cli/src/formatter.rs (hunk 2 above)
Comment (jonathantanmy) [reply cli/src/formatter.rs]: Done.

**Pull request merged**